raft/raft.go

```diff
@@ -632,7 +632,7 @@ func stepLeader(r *raft, m pb.Message) {
 			r.id, r.raftLog.lastTerm(), r.raftLog.lastIndex(), r.Vote, m.From, m.LogTerm, m.Index, r.Term)
 		r.send(pb.Message{To: m.From, Type: pb.MsgVoteResp, Reject: true})
 	case pb.MsgSnapStatus:
-		if pr.State != ProgressStateSnapshot {
+		if pr == nil || pr.State != ProgressStateSnapshot {
```

Comment:
I tried pulling out the voting check, but MsgBeat still needs to be able to run without progress.  So I've instead split the switch statement in two.  With the first set of message types that do not require that the progress exist, the second set requiring that it does exists, and the check for it in the middle.